assets/js/components/ProjectEditor.js

```diff
@@ -65,7 +65,17 @@ export function ProjectEditor (projectDescriptionCredits, programId, textFields,
 
   $(document).ready(getLanguages)
 
-  this.show = () => {
+  this.show = (navigationHandler, navigationCallback, language, showLanguageSelect, showDeleteButton, headerText) => {
+    this.navigationHandler = navigationHandler
+    this.navigationCallback = navigationCallback
```

Comment:
I find the handler and callback confusing. see the other comments

Reply:
Removed the navigation handler. I added the navigation callback later and didn't realize that the navigation handler and some other code could be moved in there. Should be fixed now